01-data-structures/05-hashes-part-2/separate_chaining/separate_chaining.rb

```diff
@@ -1,33 +1,109 @@
 require_relative 'linked_list'
 
 class SeparateChaining
-  attr_reader :max_load_factor
+  # attr_reader :max_load_factor
 
   def initialize(size)
     @max_load_factor = 0.7
+    @items = Array.new(size)
+    @numItems = 0
   end
 
   def []=(key, value)
+    newItem = Node.new(key, value)
+    newIndex = index(key, size)
+
+    if @items[newIndex]
+      @items[newIndex].add_to_tail(newItem)
+
+      # puts "item added was #{@items[newIndex]} and it was added to position #{newIndex}"
+    else
+      @items[newIndex] = LinkedList.new
+      @items[newIndex].add_to_tail(newItem)
+
+      # puts "item added was #{@items[newIndex]} and it was added to position #{newIndex}"
+    end
+
+    @numItems += 1
   end
 
   def [](key)
+    newIndex = index(key, size)
+    if @items[newIndex]
+      data = @items[newIndex].head
+      while data
+        if data.key == key
+          return data.value
+        else
+          data = data.next
+        end
+      end
+    end
   end
 
   # Returns a unique, deterministically reproducible index into an array
   # We are hashing based on strings, let's use the ascii value of each string as
   # a starting point.
   def index(key, size)
+    key.sum % size
   end
 
   # Calculate the current load factor
   def load_factor
+    resize if @numItems.fdiv(size) > @max_load_factor
+
+    @numItems.fdiv(size)
   end
 
   # Simple method to return the number of items in the hash
   def size
+    @items.length
   end
 
   # Resize the hash
   def resize
+    newLength = (@items.length * 2)
+    oldArray = @items
+    @items = Array.new(newLength)
+
+    for i in 0..oldArray.length do
+      next unless oldArray[i]
+      dataMove = oldArray[i].head
+      newIndex = index(dataMove.key, newLength)
```

Comment:
While true for this data set, you cannot assume every item in a linked list will end up in the same place. For example, `job` and `feather`. At size 12 they both land in index 3, but at 24 `job` stays at 3 while `feather` moves to 15.

https://repl.it/repls/GorgeousOrangeIberiannase

Reply:
Ok, Just trying to figure out what you're looking for. Is this a comment for future notice or do you want me to build something out to remedy it?